scripts/16-rust-testing.md

```diff
@@ -0,0 +1,647 @@
+<style>
+:root {--r-code-font: "FiraCode Nerd Font";}
+.reveal .hljs {min-height: 50%;}
+</style>
+%%
+
+# Cargo.toml
+
+```toml
+[package]
+name = "rust-testing"
+version = "0.1.0"
+edition = "2021"
+
+[build-dependencies]
+
+[dev-dependencies]
+
+[dependencies]
+```
+
+# Lint Tweaks
+
+```rust
+#![allow(dead_code)]
+#![allow(unused_variables)]
+```
+
+# Extern Crates
+
+```rust
+
+```
+
+# Imports
+
+```rust
+```
+
+# Setup
+
+```rust
+fn main() -> color_eyre::eyre::Result<()> {
+    color_eyre::install()?;
+	println!("Rust Testing");
+	assertion_test();
+	Ok(())
+}
+```
+
+%%
+
+![[rust-logo.png]]
+
+# RUST Tests Itself
+
+%% Kind of %%
+notes:
+
+Hi friends my name is Tris and this is No Boilerplate, focusing on fast, technical videos.
+
+Today we're going to talk about how to test Rust, and how this is different to testing in other popular languages.
+
+---
+
+> If debugging is the process of removing software bugs, then programming must be the process of putting them in.
+
+&mdash; Dijkstra
+
+notes:
+
+Rust is so reliable because if used correctly, whole categories of bugs are impossible to express.
+
+For the remainder of the bugs that ARE possible to express, you will indeed need to test.
+
+---
+
+| Happy Path   | Unhappy Path | Random     |
+| ------------ | ------------ | ---------- |
+| Assertations | Black Box    | QuickCheck |
+| Doctests     | White Box    | Proptest   |
+| Examples     |              | Fuzzing    |
```

Comment:
I'd like to see Integration covered, as I think it's really important. They way Martin Fowler [describes it](https://martinfowler.com/bliki/IntegrationTest.html) is the best I've come across.

There's a [longer-form version](https://martinfowler.com/articles/practical-test-pyramid.html) as well.

Reply:
nice suggestion, I assumed integration would be a whole extra video, but no harm mentioning it - good call